src/libcollections/str.rs

```diff
@@ -596,16 +748,26 @@ impl str {
         core_str::StrExt::bytes(self)
     }
 
-    /// An iterator over the non-empty substrings of `self` which contain no whitespace,
-    /// and which are separated by any amount of whitespace.
+    /// Split a string slice by whitespace.
+    ///
+    /// The iterator returned will return string slices that are sub-slices of
+    /// the original string slice, separated by any amount of whitespace.
+    ///
+    /// 'Whitespace' is defined according to the terms of the Unicode Derived
+    /// Core Property `White_Space`.
     ///
     /// # Examples
     ///
+    /// Basic usage:
+    ///
     /// ```
-    /// let some_words = " Mary   had\ta\u{2009}little  \n\t lamb";
```

Comment:
This seemed like a nice example which showed off that no just spaces were accounted for, perhaps that could stick around?